hadoop-common-project/hadoop-common/src/site/markdown/filesystem/filesystem.md

```diff
@@ -905,6 +905,27 @@ the file, then verify the invariants in the `PathHandle` using
 `getFileStatus(Path)` to implement `CONTENT`. This could yield false
 positives and it requires additional RPC traffic.
 
+
+### `copyFile(URI srcFile, URI dstFile)`
+
+Copies a file `srcFile` to another file `dstFile`.
+
+Implementations without a compliant call SHOULD throw `UnsupportedOperationException`.
+
+#### Preconditions
+
+    if not exists(FS, srcFile) : raise FileNotFoundException
+
+    if not exists(parentFolder(dstFile)) : raise [IllegalArguementException]
+
+    if isDirectory(srcFile) : raise [IllegalArguementException]
+
+    if exists(dstFile) : raise FileAlreadyExistsException
```

Comment:
why? Because with s3 I you don't get an atomic check anyway...it's simplest just to do the copy and say "yes, we overwrite".